integration_tests/meta_eval_wmt_da_test.py

```diff
@@ -39,94 +43,170 @@ def test_da_cs_en(self):
         sys_info = processor.process(metadata, data)
         self.assertGreater(len(sys_info.results.analyses), 0)
         self.assertGreater(len(sys_info.results.overall), 0)
+        self.assertAlmostEqual(
+            sys_info.results.overall[0]["SegKtauCorr"].value, -0.0169, 3
+        )
+
+
+class MetaEvalNLGInvalidValueTest(unittest.TestCase):
+
+    true_data = [[1, 2, 3, 4, 5], [2, 1, 4, 5, 2], [5, 4, 3, 2, 1]]
+    pred_data = [[2, 1, 3, 4, 5], [2, 4, 5, 5, 2], [5, 3, 4, 2, 1]]
+
+    def test_illegal_correlation_type(self) -> None:
+
+        with self.assertRaisesRegex(
+            ValueError, r"^The correlation " r"function illegal hasn't been supported"
+        ):
+            nlg_corr_config = CorrelationNLGConfig(
+                group_by="sample", correlation_type="illegal"
+            )
+            corr_metric = narrow(CorrelationNLG, nlg_corr_config.to_metric())
+            stats = corr_metric.calc_stats_from_data(self.true_data, self.pred_data)
+            stats_arr = corr_metric.aggregate_stats(stats)
+            val = corr_metric._calc_metric_from_aggregate_single(stats_arr)
+            self.assertAlmostEqual(val, 0.8162952, 3)
```

Comment:
tip: If the test uses `assertRaises`, it is recommended to wrap only one statement to check if a particular statement should raise the error and any other statements can be processed appropriately.

There are two positions that the targeted error will occur: `calc_stats_from_data` and `aggregate_stats`. They should be tested separately.

```suggestion
    def test_illegal_correlation_type_calc_stats_from_data(self) -> None:
        with self.assertRaisesRegex(ValueError, r"^The correlation function"):
        nlg_corr_config = CorrelationNLGConfig(
            group_by="sample", correlation_type="illegal"
        )
        corr_metric = narrow(CorrelationNLG, nlg_corr_config.to_metric())
        
        with self.assertRaisesRegex(ValueError, r"^The correlation function"):
            corr_metric.calc_stats_from_data(self.true_data, self.pred_data)
            
    def test_illegal_correlation_type_aggregate_stats(self) -> None:
        with self.assertRaisesRegex(ValueError, r"^The correlation function"):
        nlg_corr_config = CorrelationNLGConfig(
            group_by="sample", correlation_type="illegal"
        )
        corr_metric = narrow(CorrelationNLG, nlg_corr_config.to_metric())
        stats = SimpleMetricStats(np.zeros(...))
        
        with self.assertRaisesRegex(ValueError, r"^The correlation function"):
            corr_metric.aggregate_stats(stats)
```